### DISCUSSION TITLE
um front pode se tornar um UX/UI Designer?

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#723] [Discussão]
Pessoal, eu sou front a 7 anos, e ainda pretendo continuar e ser fullstack também talvez, gosto muito de codar, mas quando vejo projetos de design, desde a concepção, até as animações, as UI mesmos, meus olhos saltam, quando olho um algoritmo, ou um código acho bacana, mas não sinto aquela "saltada" ou pulo no coração.

Meu problema que eu sou muito generalista ainda, e tenho noção de tudo ,e especialista em nada.E também não fiz faculdade. Vocês acham que um profissional com 27 anos, consegue mesmo estudando javascript , node e etc, aos poucos ir criando um portfolio com design e ir pra essa área? ( Mesmo sabendo que o mercado é bem menor ) 

Conhecem exemplos de programadores que se tornaram UX/UI designers?  Eu, gosto de pensar no produto todo aliás, mas vejo como dois caminhos, um de abstração visual e o outro de abstração lógica, e me vejo muito mais forte na abstração visual.

**Comment by rubenmarcus** [comment]
ou existem mesmo profissionais generalistas que fazem projetos, desde o UX , até a UI e a implementação da UI e depois o sistema?

**Comment by kvnol** [comment]
Por favor, @mjcoffeeholick e @talitapagani

♥

**Comment by rubenmarcus** [comment]
@buKinoshita https://bukinoshita.io/ fucei seu portfolio uma vez e vi que é ux designer e front-end engineer, vc poderia falar algo mais sobre?

**Comment by bukinoshita** [comment]
Opa @rubenmarcus! Vou tentar te ajudar contando um pouco sobre mim e um pouco sobre sua situação.

**tl;dr**: Sim! 🎉 


## Table of contents
  * [👩‍🎨  Querer virar ux designer e back-end](#querer-virar-ux-designer-e-back-end)
  * [📃 Um pouco da minha história](#um-pouco-da-minha-historia)
  * [🛠 Tools](#tools)
  * [🍻 Conclusão](#conclusao)

_edit: acho que table of contents não funciona(?)_

## Querer virar ux designer e back-end

É totalmente possível, mas tem que tomar cuidado com o `master of none`, né? Você pode aprender quantas coisas você quiser, mas tem que ter em mente que, talvez, você não seja bom em tudo.

Meu modo de pensar e o que eu acho melhor, pelo menos pra mim, é ter `[foco1, foco2, foco3, ...]`. Onde você passa mais tempo no `foco1` do que no `foco2` e assim por diante. Ou seja, você provavelmente será melhor no `foco1` do que no `foco2`. Ao invés de você tentar fazer `[foco1, foco1, foco1, ...]` e no fim não ser bom em nada, mas você sabe um pouco de cada coisa.
<br/>
> Conhecem exemplos de programadores que se tornaram UX/UI designers?

Eu não conheço, porque não é algo que eu fico procurando. Mas com certeza devem existir, da mesma forma que um designer quis focar em programação, um programador quis forcar em business, etc...


## Um pouco da minha história

Eu comecei com front-end (14yo), mas tipo era só umas linhas de HTML que eu sabia, não montava nada ainda. Eu comecei porque sempre tive curiosidade de saber como montar um site.

Depois de um tempo eu comecei ir pro design (15yo), mas era design gráfico. Eu passava meu tempo brincando no Photoshop, fazendo photomanipulation e uns freelas.

![photomanipulation](https://user-images.githubusercontent.com/6929565/30456679-9d606fa2-9972-11e7-9402-ceae4c526ddc.jpg)
_(photomanipulation que fiz anos atrás por diversão :P)_

Certo dia, meu irmão que é back-end developer mencionou sobre uma empresa foda que  era focada em UX Design. Eu curti e me aprofundei para aprender UX.

Então ao invés de ficar fazendo freela/photomanipulations eu comecei a brincar com UI/UX no Photoshop e entender como funciona. Algo mais voltado pra web/mobile.

![ui](https://user-images.githubusercontent.com/6929565/30456933-975ae01e-9973-11e7-8f4e-89fad8a4bb43.jpg)
_(London official visitor guide / projetinho que fiz uns anos atrás)_

E do nada, um dia eu pensei "vou codar um site". Passei o domingo inteiro fazendo um site e lembro até hoje o quão divertido foi. Depois desse dia, eu foquei totalmente em front-end, mas design sempre seguiu do meu lado. Sempre revezava no que focar.

![tinder-final](https://user-images.githubusercontent.com/6929565/30457370-2848dbca-9975-11e7-8a93-c146a3ac6f33.gif)
_(tinder login redesign)_

Nisso, a web foi evoluindo e eu aprofundando tanto no code, como no design. Aprendi animações, aprendi JS e suas libs (react, vue, angular, jquery...) etc.

![edit-tweet](https://user-images.githubusercontent.com/6929565/30457155-5454e278-9974-11e7-824d-d4f46dfa79e8.gif)
_(edit tweet concept)_

Larguei meu trampo no começo do ano pra focar em open source, criar business e projetos e aprender mais sobre design, front e back.

Eu tinha um conhecimento básico sobre back-end, nesse tempo eu dei uma aprofundada e lancei vários projetos que tinha em mente just for fun.

O que eu gosto no design, é que ele me ajuda a resolver problemas.

![nice-7921](https://user-images.githubusercontent.com/6929565/30457454-62136cc6-9975-11e7-9ad6-a6f076bf340c.gif)
_(shhhowr, um concept que fiz pra explorar interligência artificial e um problema que eu estava tendo em casa)_


## Tools

- [Dribbble](https://dribbble.com/) — Aprender e se inspirar
- [Sketch App](https://www.sketchapp.com/) — Design
- [Principle](http://principleformac.com/) — Animação


### Talvez te ajude...

- [awesome-ux](https://github.com/netoguimaraes/awesome-ux)
- [awesome-design](https://github.com/gztchan/awesome-design)
- [uxpin e-books](https://www.uxpin.com/knowledge.html)
- [invision](https://www.invisionapp.com/blog)


## Conclusão

Você pode sim focar em várias coisas, mas tenha em mente que você (talvez) não seja bom em tudo. Por exemplo, eu manjo design, front e back. Meu foco e o que eu dou prioridade é o front, então eu sou melhorzinho. Depois vem o design e depois o back.

Ps: It's never too late!
Ps2: O que eu to mostrando aqui é mais UI. A parte de UX é mais research da minha parte, que eu ainda vou escrever no meu blog sobre os concepts que eu fiz. :P
Ps3: Se quiser me chamar pra trocar ideia, feel free.

**Comment by talitapagani** [comment]
O @bukinoshita já te deu uma resposta maravilhosa!

Sim, é possível, isso foi o meu caminho também.

Eu sou formada em ciência da computação, com mestrado também na área e mesmo assim fui migrando aos poucos de programação para UX. Desde 2005, eu trabalhava tanto com código quanto com design (também conhecido como o antigo web designer) e, ao longo dos anos, fui migrando e incorporando mais habilidades de UX e design de interação.

Não é incomum esse movimento. Quando você está codificando um layout, você também está codificando as interações que foram projetadas pelos designers, afinal, [interface é código também](https://pt.slideshare.net/talitapagani/interface-cdigo-aprimorando-a-experincia-do-usurio-no-front-e-no-backend). Nesse momento, você percebe que certas interações podem não funcionar e naturalmente se envolve em pensar em como aprimorar a forma de apresentar, transitar entre ações, exibir conteúdo, dar feedback aos usuários, etc.

> Você pode sim focar em várias coisas, mas tenha em mente que você (talvez) não seja bom em tudo. Por exemplo, eu manjo design, front e back. Meu foco e o que eu dou prioridade é o front, então eu sou melhorzinho. Depois vem o design e depois o back.

Concordo com essa afirmação do Bu. Eu também tenho conhecimento na stack completa, já fiz projetos desde o planejamento até o back-end e, na empresa onde estou hoje, há projetos em que eu atuo desde o levantamento de requisitos até o front-end e há projetos em que eu estou só em uma das frentes. E, sim, é possível entregar projetos de qualidade dessa forma se fosse consegue trabalhar essa multidisciplinaridade. Mas também é interessante que você identifique seu ponto forte e dê maior foco nele. No meu caso, eu demorei a perceber que realmente meu forte é mais a parte de UX (pesquisa de campo, design da interação, user flow, protótipos, testes com usuários, etc.) e então eu tenho fortalecido essas habilidades, sem deixar de lado os aprendizados de front-end.

> Vocês acham que um profissional com 27 anos, consegue mesmo estudando javascript , node e etc, aos poucos ir criando um portfolio com design e ir pra essa área? ( Mesmo sabendo que o mercado é bem menor )

R: Sim! Não se prenda a idade, você é novo e consegue ir migrando para a área desejada e criar um portfolio. O mercado de UX não é tão pequeno, está aquecido e precisando de profissionais. Uma dica preciosa é: entenda bem de negócios. Designers de experiência de usuário não devem atender somente aos usuários como diz o nome, mas a diferentes partes interessadas e, para isso, é preciso ter conhecimento de "business" também, faz uma grande diferença.

Boa sorte e estou à disposição para trocar ideias!

**Comment by rubenmarcus** [comment]
muito bom

**Comment by cristofersousa** [comment]
Veja se essa vaga se enquadra ao que você busca.

[Vaga  UX/UI Egenius ] (https://docs.google.com/document/d/1DnLH5jT9ZQQxYD6jMaJ0mTGpgFDgH1T4uN2GcnymJm4/edit)

**Comment by lucianobarauna** [comment]
Sensacional as repostas @bukinoshita e @talitapagani 
O blog da @talitapagani eu acompanhava mais não sabia que era fromada em ciência eu jurava que era em design rs

Bom @rubenmarcus um pouco de experiência. 

Como já mencionei aqui em alguns post eu sou front-end formado em design esolhi essa faculdade justamente para pegar mais esse feeling, gastar da preocupação com o usário e de ter uma boa interface.

Já trabalho a 6 anos na área e tenho meus 29 aninhos. Atualmente estou querendo fazer uma faculdade de TI EAD em Análise de Sistemas para pegar mais conhecimento técnico e entender algumas coisas.

Amo front-end gosto até mais do que design afinal foi onde eu comecei realmente mais hoje a faculdade em TI talvez seje preciso eu fazer para conseguir vagas legais porquê infelizmente vagas boas para front geralmente o na descrição da vaga sempre vem `Faculdade em TI ou realacionadas` e isso para uma seleção de RH onde ninguém tem noção pode ser critério de eliminação como já foi para mim.

Então minha preocupação hoje de fazer TI é ter mais uma chance em vagas e por gostar da área de front.

Espero que a experiência ajude ;-)

**Comment by rubenmarcus** [comment]
@cristofersousa  Se enquadra, mas eu nao tenho formação em Design e também não tenho conhecimentos em testes. Eu estou tirando uns meses até Janeiro pra estudar, tenho um curso de Interation Design que estou terminando e mais uns na área de front que quero estudar pra voltar pro mercado, no momento só estou com freelas pequenos, e estudando mesmo.

**Comment by rubenmarcus** [comment]
@lucianobarauna  Acho que a faculdade conta, mas ter experiência e ser um bom programador conta mais. Tem diversos não formados, auto-ditatas que conseguiram.

**Comment by lucianobarauna** [comment]
Sim lógico...e concordo muito bom isso. Só que talvez em algum momento isso pode fazer falta.

**Comment by bukinoshita** [comment]
@rubenmarcus eu trampei dois anos em Toronto e depois NY, mas ambos foram remotos. Eu não tenho faculdade e nunca precisei. Mas poucos tempos atrás aprendi que faculdade vai te ajudar a pegar um visa se quiser. O mais comum é o H-1B, que diz que precisa ter 12 pontos no total pra poder aplicar. Cada ano de faculdade são 3 pontos e cada ano trabalhando na área conta como 1 ponto. Ou seja, como eu não tenho facul, eu preciso trabalhar 12 anos na área pra poder tirar um visto se quiser trampar fora. Claro que isso não é um regra exata se você conseguir comprovar que manja, mas eles são rígidos. Isso é US e Canada. Outros países não manjo.

Só uma informação a mais :)

**Comment by danielrochadev** [comment]
Good!

**Comment by rubenmarcus** [comment]
@bukinoshita , @talitapagani , @lucianobarauna 

Revisitando esse tópico. Hoje eu sou Front Senior aqui no Brasil , mas ainda tenho vontade em trabalhar em outras frentes como UI/UX, e sem faculdades. 
vocês acham mais útil fazer uma facul primeiro relacionado a uma área de programação ( Sistemas para Internet, ADS, Ciencia da Computação)

e depois uma em relação a design ou multimidia ( Produção Multimidia, Design , Artes Visuais )?

Hoje eu ganho relativamente bem como Front, e migrar de area assim, pra mim é complicado porque front paga minhas contas, ap ensão do meu filho, eu não posso ter incertezas sabe?

Mas eu me vejo muito no que o Bu falou, e preciso estudar e fazer algo relacionado a area. O que voces indicam ou fariam no meu lugar? Eu quero fazer uma EAD para tentar conseguir mais facil visto pra ir pra fora, e aqui no Brasil tb dar uma condição melhor e destaque.

**Comment by talitapagani** [comment]
@rubenmarcus mesmo UX sendo muito amplo, eu penso que um curso em Design Digital pode ser mais proveitoso para te fornecer uma base formal relacionada a UX. Eu fiz Ciência da Computação e, olha, dificilmente tem algo relacionado. Disciplinas de Interação Humano-Computador (IHC) nem sempre constam no currículo e quando constam, são optativas.
